deployment/terraform/batch_pool/main.tf

```diff
@@ -1,6 +1,7 @@
 resource "azurerm_batch_pool" "batch_pool" {
 
   name                = var.name
+  metadata            = {}
```

Comment:
This and `task_scheduling_policy` are just to get the terraform to match what's set on the pools.